src/mongo/s/commands/cluster_count_cmd.cpp

```diff
@@ -71,29 +71,19 @@ class ClusterCountCmd : public ErrmsgCommandDeprecated {
         out->push_back(Privilege(parseResourcePattern(dbname, cmdObj), actions));
     }
 
-    bool errmsgRun(OperationContext* opCtx,
-                   const std::string& dbname,
-                   const BSONObj& cmdObj,
-                   std::string& errmsg,
-                   BSONObjBuilder& result) override {
-        const NamespaceString nss(parseNs(dbname, cmdObj));
-        uassert(ErrorCodes::InvalidNamespace,
-                str::stream() << "Invalid namespace specified '" << nss.ns() << "'",
-                nss.isValid());
-
-        long long skip = 0;
+    bool run(OperationContext* opCtx,
+             const std::string& dbname,
+             const BSONObj& cmdObj,
+             std::string& errmsg,
+             BSONObjBuilder& result) override {
 
-        if (cmdObj["skip"].isNumber()) {
-            skip = cmdObj["skip"].numberLong();
-            if (skip < 0) {
-                errmsg = "skip value is negative in count query";
-                return false;
-            }
-        } else if (cmdObj["skip"].ok()) {
-            errmsg = "skip value is not a valid number";
-            return false;
+        const bool isExplain = true;
+        auto request = CountRequest::parseFromBSON(dbname, cmdObj, isExplain);
+        if (!request.isOK()) {
```

Comment:
Might as well just do

`uassertStatusOK(request.getStatus())`

We're trying to get a little more throw-y and a little less Status-y when it comes to propagating errors.